File: Previous Contest/3550. Smallest Index With Digit Sum Equal to Index/Solution.py
from typing import List

class Solution:
    def smallestIndex(self, nums: List[int]) -> int:
        # N : len(nums), M : Max len of digit of nums[i]
        # Time Complexity: O(N * M)
        # Space Complexity: O(1)
        def digitSum(num):
            summ = 0
            while num != 0:
                summ += (num % 10)
                num //= 10
            return summ
        
        n = len(nums)
        for i in range(n):
            if i == digitSum(nums[i]):
                return i
        return -1

File: Previous Contest/3550. Smallest Index With Digit Sum Equal to Index/Solution.java
class Solution {
    private int digitSum(int num){
        int summ = 0;
        while (num != 0){
            summ += num % 10;
            num /= 10;
        }
        return summ;
    }
    public int smallestIndex(int[] nums) {
        // N : len(nums), M : Max len of digit of nums[i]
        // Time Complexity: O(N * M)
        // Space Complexity: O(1)
        int n = nums.length;
        for (int i=0; i<n; i++){
            if (i == digitSum(nums[i])) return i;
        }
        return -1;
    }
}